### PR TITLE
Two-level flag documentation

### PR DESCRIPTION
**Pull request by ElifUskuplu**
Hi, I've prepared a documentation about ``two-level`` flag with Mike Shulman. It's based on my experience about the [2LTT library ](https://github.com/ElifUskuplu/2LTT-Agda) I created.

### CODE REVIEW

**Comment by andreasabel** [comment]
Thanks!
Rendering here: https://agda--6321.org.readthedocs.build/en/6321/language/two-level.html

CI failure when building HTML:
```
doc/user-manual/language/two-level.lagda.rst: WARNING: document isn't included in any toctree
```
Looks like you need to add the new file to [`language/index.rst`](https://github.com/agda/agda/blob/master/doc/user-manual/language/index.rst).

Note: you can locally test the user manual by typing (in the root):
```
make user-manual-html
make user-manual-pdf
make user-manual-test
```

File: doc/user-manual/language/two-level.lagda.rst [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+
+*********************
+Two-Level Type Theory
+*********************
+
+Basics
+------
+
+Two-level type theory (2LTT) refers to versions of Martin-Lof type
```
Comment (andreasabel) [inline doc/user-manual/language/two-level.lagda.rst]: Martin Löf

File: doc/user-manual/language/two-level.lagda.rst [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+"outer" level that validates uniqueness of identity proofs.
+
+Since version 2.6.2, Agda enables 2LTT with the ``--two-level`` flag.
+The two levels are distinguished with two hierarchies of universes:
+the usual universes ``Set`` for the inner level, and a new hierarchy
+of universes denoted ``SSet`` (for "strict sets") for the outer level.
+
+.. note::
+   The types in ``SSet`` have various names in the literature. They
+   are called `non-fibrant types` in [`HTS (2017)
```
Comment (andreasabel) [inline doc/user-manual/language/two-level.lagda.rst]: Delete the brackets [...] around these citations.

File: doc/user-manual/language/two-level.lagda.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+each universe ``Set a`` becomes a subtype of ``SSet a``.  In this case
+we can instead define the coercion ``c`` to be the identity function::
+
+  c' : Set → SSet
+  c' A = A
+
+and replace the coercions ``↑`` and ``↓`` with the identity function.
+However, this combination currently allows some functions to be
+defined that shouldn't be allowed; see issue `#5761
+<https://github.com/agda/agda/issues/5761>` for details.
```
Comment (andreasabel) [inline doc/user-manual/language/two-level.lagda.rst]: This link is broken, I think an underscore is needed after the final backtick.

File: doc/user-manual/language/two-level.lagda.rst [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  data ℕˢ : SSet where
+    zeroˢ : ℕˢ
+    succˢ : ℕˢ → ℕˢ
+
+  ℕˢ-to-ℕ : ℕˢ → ℕ
+  ℕˢ-to-ℕ zeroˢ = zero
+  ℕˢ-to-ℕ (succˢ n) = succ (ℕˢ-to-ℕ n)
+
+but not vice versa.  (Agda does currently allow mapping from the empty
+``SSet`` to the empty ``Set``, but this is a bug.)
```
Comment (andreasabel) [inline doc/user-manual/language/two-level.lagda.rst:100]: Why is this a bug?

**Review by andreasabel — COMMENTED** [review]
Thanks for this nice article!

File: doc/user-manual/language/two-level.lagda.rst (hunk 4 above)
Comment (ElifUskuplu) [reply doc/user-manual/language/two-level.lagda.rst:100]: It is explained [ here ](https://github.com/agda/agda/issues/5761#issuecomment-1154239427) .

**Comment by ElifUskuplu** [comment]
Thanks for your consideration

File: doc/user-manual/language/two-level.lagda.rst (hunk 4 above)
Comment (andreasabel) [reply doc/user-manual/language/two-level.lagda.rst:100]: Could you replace "bug" by "disputed" and link to that issue comment?

**Review by andreasabel — APPROVED** [review]

File: doc/user-manual/language/two-level.lagda.rst (hunk 4 above)
Comment (andreasabel) [reply doc/user-manual/language/two-level.lagda.rst:100]: I'll do it.

**Comment by andreasabel** [comment]
I cannot add commits to this PR, so I reopened it as:
- #6340

**Pull request closed without merging**